sprockets/mixins/http/__init__.py

```diff
@@ -6,50 +6,248 @@
+        """Return the parsed link header if it was set, returning a list of
+        the links as a dict.
+
+        :rtype: list(dict()) or None
+
+        """
+        if not self._responses:
+            return None
+        if 'Link' in self._responses[-1].headers:
+            links = []
+            for l in headers.parse_link(self._responses[-1].headers['Link']):
+                link = {'target': l.target}
+                link.update({k: v for (k, v) in l.parameters})
+                links.append(link)
+            return links
+
+    @property
+    def ok(self):
+        """Returns `True` if the response status code was between 200 and 399.
+        Returns `False` if no responses were received or the response status
+        code was >= 400.
+
+        :rtype bool
+
+        """
+        if not self._responses:
+            return False
+        return 200 <= self._responses[-1].code < 400
+
+    @property
+    def raw(self):
+        """Return the raw tornado HTTP Response object
+
+        :rtype: tornado.httpclient.HTTPResponse
+
+        """
+        if not self._responses:
+            return None
+        return self._responses[-1]
+
+    def _decode(self, value):
+        """Decode bytes to UTF-8 strings as a singe value, list, or dict.
+
+        :param mixed value: The value to decode
+        :rtype: mixed
+
+        """
+        if isinstance(value, list):
+            return [self._decode(v) for v in value]
+        elif isinstance(value, dict):
+            return {self._decode(k): self._decode(v)
+                    for k, v in value.items()}
+        elif isinstance(value, bytes):
+            return value.decode('utf-8')
+        return value
+
+    def _deserialize(self):
+        """Try and deserialize a response body based upon the specified
+        content type.
+
+        :rtype: mixed
+
+        """
+        if not self._responses or not self._responses[-1].body:
+            return None
+        if 'Content-Type' not in self._responses[-1].headers:
+            return self._responses[-1].body
+        try:
+            content_type = algorithms.select_content_type(
+                [headers.parse_content_type(
+                    self._responses[-1].headers['Content-Type'])],
+                AVAILABLE_CONTENT_TYPES)
+        except errors.NoMatch:
+            return self._responses[-1].body
+
+        if content_type[0] == CONTENT_TYPE_JSON:
+            return self._decode(
+                self._json.loads(self._decode(self._responses[-1].body)))
+        elif content_type[0] == CONTENT_TYPE_MSGPACK:  # pragma: nocover
+            return self._decode(
+                self._msgpack.unpackb(self._responses[-1].body))
+
+    def _error_message(self):
+        """Try and extract the error message from a HTTP error response.
+
+        :rtype: str
+
+        """
+        body = self._deserialize()
+        return body.get('message', body) if isinstance(body, dict) else body
 
 
 class HTTPClientMixin:
```

Comment:
How much work would be involved with making a standalone HTTP client with separate mixins for Tornado and Rejected? I could see some using the bare client in one-off scripts - particularly for error retry.

Reply:
Probably not too hard, but more than I'm willing to do for this PR, I've got a fix for the analytics-consumers waiting on it.